source/common/http/utility.h

```diff
@@ -300,9 +301,25 @@ absl::flat_hash_map<std::string, std::string> parseCookies(const RequestHeaderMa
  * Parse a particular value out of a set-cookie
  * @param headers supplies the headers to get the set-cookie from.
  * @param key the key for the particular set-cookie value to return
- * @return std::string the parsed set-cookie value, or "" if none exists
+ * @return absl::string_view the parsed set-cookie value, or "" if none exists
```

Comment:
Here and below please document the lifetime of the returned string views 

More generally I'd see if there are any concerns about this interface switch, given that a std::string ends up being a bit safer than a string_view, cc @alyssawilk @mattklein123

Reply:
Maybe the safest option for now is indeed to keep `std::string`. I will revert back to that API.

Reply:
Done in https://github.com/envoyproxy/envoy/pull/21496/commits/b7f34796cf5bbe216a159745a0a86cc49a0d2268